### DISCUSSION TITLE
Specify the registry for a module

### DISCUSSION THREAD

**Discussion by edoardocavazza** [npm/feedback#87] [CLI]
Hello!

Our team use to publish private packages the NPM registry, but for some packages we would like to use the Github Package Registry. The organization name is the same for both NPM and Github, so we can not just specify the registry using the scope name. Can we instruct *where* the npm cli should download a list of packages? It would be ok to use a specific syntax in the package.json or in a npmrc file.

Thank you!

**Comment by MylesBorins** [comment]
@edoardocavazza thanks for bringing this up. Was just discussing something similar with a colleague today. We are thinking through what including registry in the specifier could look like and the implications of doing so.

One question on my mind would be would the special syntax be only in the package.json or also in the module itself. Since there could potentially be duplicated scopes across registries it might be beneficial to be explicit, but that also gets very verbose very quickly.

In your use case you are using the npm registry for public packages and github packages for private packages. Would you ever be looking at using a third registry?

**Reply by edoardocavazza** [reply]
Actually, we use NPM registry for both public and private packages, but it would be not a problem to migrate or duplicate private modules in the GH registry. I would prefer to not use a third registry in order to avoid security issues 😔

**Comment by ljharb** [comment]
I think https://github.com/npm/rfcs/issues/275 would meet this use case.

**Reply by edoardocavazza** [reply]
It seems interesting! thank you!
